lib/inputs/redis-increments.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const logger = require('../logger');
+const redis = require('../redis');
+
+/**
+ * Increment redis counts
+ */
+module.exports = class RedisIncrements {
+
+  constructor(records) {
+    this._records = (records || []).filter(r => this.check(r));
+  }
+
+  check(record) {
+    return (record.type === 'download' || record.type === 'impression')
+        && (record.feederPodcast || record.feederEpisode)
+        && !record.isDuplicate;
+  }
+
+  insert() {
+    if (this._records.length == 0) {
+      return Promise.resolve([]);
+    } else {
+      let keysToFields = {}, totalIncrements = 0;
+
+      this._records.forEach(r => {
+        if (r.feederPodcast) {
+          totalIncrements += this.addKeys(keysToFields, this.getPodcastKeys(r), r.feederPodcast);
+        }
+        if (r.feederEpisode) {
+          totalIncrements += this.addKeys(keysToFields, this.getEpisodeKeys(r), r.feederEpisode);
+        }
+      });
+
+      let increments = Object.keys(keysToFields).map(k => this.doIncrements(k, keysToFields[k]));
+      let expires = Object.keys(keysToFields).map(k => this.doExpire(k));
+      return Promise.all(increments.concat(expires)).then(replies => {
+        return [{count: totalIncrements, dest: this.redisHostName()}];
+      });
+    }
+  }
+
+  doIncrements(key, fieldsToCounts) {
+    return Promise.all(Object.keys(fieldsToCounts).map(field => {
+      return redis.increment(key, field, fieldsToCounts[field]);
+    }));
+  }
+
+  doExpire(key) {
+    return redis.expire(key);
+  }
+
+  getPodcastKeys(record) {
+    if (record.type === 'download') {
+      return redis.podcastDownloads(record.timestamp);
+    } else {
+      return redis.podcastImpressions(record.timestamp);
+    }
+  }
+
+  getEpisodeKeys(record) {
+    if (record.type === 'download') {
+      return redis.episodeDownloads(record.timestamp);
+    } else {
+      return redis.episodeImpressions(record.timestamp);
+    }
+  }
+
+  addKeys(map, keys, id) {
```

Comment:
This is a little confusing ... but it's rolling up the podcasts/episodes to increment, so I can just call one `HINCRBY 2` instead of two `HINCRBY 1`.  It will eventually look something like:

```javascript
map = {
  'downloads.podcasts.HOUR.2017-07-15T14:00:00Z' => {
    1234: 3
    5678: 1
  },
  'downloads.podcasts.15MIN.2017-07-15T14:00:00Z' => {
    1234: 1,
    5678: 1
  }
  'downloads.podcasts.15MIN.2017-07-15T14:15:00Z' => {
    1234: 2
  }
}
```